sotodlib/io/g3tsmurf_utils.py

```diff
@@ -0,0 +1,195 @@
+"""Utility functions for interacting with level 2 data and g3tsmurf
+"""
+
+import os
+import numpy as np
+import logging
+
+from sotodlib.io.load_smurf import load_file
+from sotodlib.io.g3tsmurf_db import Observations, Files
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_obs_folder(obs_id, archive):
+    """
+    Get the folder associated with the observation action. Assumes
+    everything is following the standard suprsync formatting.
+    """
+    session = archive.Session()
+    obs = session.query(Observations).filter(Observations.obs_id == obs_id).one()
+    session.close()
+    
+    return os.path.join(
+        archive.meta_path, 
+        str(obs.action_ctime)[:5], 
+        obs.stream_id, 
+        str(obs.action_ctime)+"_"+obs.action_name,
+    )
+    
+def get_obs_outputs(obs_id, archive):
+    """
+    Get the output files associated with the observation action. Assumes
+    everything is following the standard suprsync formatting. Returns 
+    absolute paths since everything in g3tsmurf is absolute paths.
+    """
+    path = os.path.join(
+        get_obs_folder(obs_id,archive),
+        "outputs"
+    )
+    
+    if not os.path.exists(path):
+        logger.error(f"Path {path} does not exist, how does {obs.obs_id} exist?")
+    return [os.path.join(path,f) for f in os.listdir(path)]
+
+
+def get_obs_plots(obs_id, archive):
+    """
+    Get the output plots associated with the observation action. Assumes
+    everything is following the standard suprsync formatting. Returns 
+    absolute paths since everything in g3tsmurf is absolute paths.
+    """
+    path = os.path.join(
+        get_obs_folder(obs_id,archive),
+        "plots"
+    )
+    
+    if not os.path.exists(path):
+        logger.error(f"Path {path} does not exist, how does {obs.obs_id} exist?")
+    return [os.path.join(path,f) for f in os.listdir(path)]
+
+def get_batch( 
+    obs_id, 
+    archive,
+    ram_limit=None, 
+    n_det_chunks=None,
+    n_samp_chunks=None, 
+    n_dets=None, 
+    n_samps=None, 
+    det_chunks=None, 
+    samp_chunks=None,
+    test = False,
+    load_file_args={},
+):
+    """A Generator to loop through and load AxisManagers of sections of
+    Observations. When run with none of the optional arguments it will default
+    to returning the full observations. Some arguments over-write others as
+    described in the docstrings below.
```

Comment:
Though ideally analysis scripts will pull this data from the axis-manager, I think the loop-order of the `dets` and `samps` axes should be included here